pkg/cmd/grafana-cli/services/api_client.go

```diff
@@ -59,7 +65,15 @@ func (client *GrafanaComClient) DownloadFile(pluginName, filePath, url string, c
 			client.retryCount++
 			if client.retryCount < 3 {
 				logger.Info("Failed downloading. Will retry once.")
-				content, err = client.DownloadFile(pluginName, filePath, url, checksum)
+				err = tmpFile.Truncate(0)
+				if err != nil {
+					panic(err)
```

Comment:
Are you sure a panic here will be recovered? You mean at line 64? I could be overlooking something, but to me it seems that if you panic here, the stack would unwind out of the current function (`GrafanaComClient.DownloadFile`).

I know it was like that before your PR, but the whole retry on panic logic looks strange to me. Why retry only on panics, and not on normal errors?

Maybe in the context of this PR, if truncate or seek fails it's better to just log the error and return (i.e. give up on retrying)?

Reply:
Right; I thought about line 64 but I guess I was wrong (it recovers from panics thrown in the main function, but not in this defered function). I've replaced it by a simple return.

And I agree that this whole retry mechanism looks fishy; I suppose that network errors would cause an error, not a `panic`. Maybe another PR?

Reply:
Thanks for rectifying that! I think another PR might be in order, since the retry on panic mechanism looks fishy, and difficult to understand too. I don't know the history of the code though, so difficult for me to say alone.